### DISCUSSION TITLE
Credentials for Heroku deployment

### DISCUSSION THREAD

**Discussion by ej-codex** [saleweaver/python-amazon-sp-api#170] [General]
I am trying to deploy an app on Heroku and keep running into authentication issues. The problem is that since Heroku does not have a file system, I cannot upload credentials.yml and for some reason, adding credentials to environment variables on Heroku does not help. Has anyone had experience with Heroku deployments over here? If so, how do you go about credentials on Heroku? 
Thanks!

**Comment by saleweaver** [comment]
How did you set which environment variables exactly?

**Reply by ej-cdx** [reply]
I have all 5 (below) in Heroku Config Vars, each one as a separate variable. I also attached a couple screenshots for reference. 

LWA_APP_ID
LWA_CLIENT_SECRET
SP_API_ACCESS_KEY
SP_API_REFRESH_TOKEN
SP_API_SECRET_KEY

Here is the error message I get: 

`2021-06-24T04:35:22.526710+00:00 app[worker.1]: sp_api.base.credential_provider.MissingCredentials: Neither environment variables nor a config file were found. Please set the correct variables, or use a config file (credentials.yml). See https://confuse.readthedocs.io/en/latest/usage.html#search-paths for search paths.`
<img width="1260" alt="Screen Shot 2021-06-24 at 1 50 33 PM" src="https://user-images.githubusercontent.com/73048600/123331081-a7a9da00-d4f3-11eb-9806-c63a1c8ac43c.png">
<img width="1233" alt="Screen Shot 2021-06-24 at 1 51 17 PM" src="https://user-images.githubusercontent.com/73048600/123331084-a8427080-d4f3-11eb-9182-1b2c3734faf0.png">

**Reply by saleweaver** [reply]
The error message is a bit confusing, I should make that more exact if there are only some missing variables. You're missing the SP_API_ROLE_ARN variable.

**Reply by ej-codex** [reply]
Wow, such a shame to miss that. Thank you, it works flawlessly!!
